### ISSUE TITLE
[analyzer] false-positive garbage value warning with multi-demensional constant array access

### ISSUE THREAD

**Issue by ueno** [llvm/llvm-project#53518]
With clang version 13.0.0 (Fedora 13.0.0-3.fc35), I got the following warning when compiling under scan-build:
```console
gost/kuznyechik.c:112:33: warning: The right operand of '^' is a garbage value [core.UndefinedBinaryOperatorResult]
  memcpy(t, &kuz_table[0][(b[0] ^ c[0]) * 16], KUZNYECHIK_BLOCK_SIZE);
                                ^ ~~~~
1 warning generated.
```
The code path is as follows:
1. `subkey` is [called](https://gitlab.com/gnutls/gnutls/-/blob/7f3ea9acc2a2da1d697d423514dfa9c3b56acb15/lib/nettle/gost/kuznyechik.c#L181) with the 3rd argument `i` set to 0
2. `LSX` is [called](https://gitlab.com/gnutls/gnutls/-/blob/7f3ea9acc2a2da1d697d423514dfa9c3b56acb15/lib/nettle/gost/kuznyechik.c#L161) with the 3rd argument `c` pointing to `kuz_key_table[i + 2]`
3. The above [line](https://gitlab.com/gnutls/gnutls/-/blob/7f3ea9acc2a2da1d697d423514dfa9c3b56acb15/lib/nettle/gost/kuznyechik.c#L112) is hit

`kuz_key_table` is [defined](https://gitlab.com/gnutls/gnutls/-/blob/7f3ea9acc2a2da1d697d423514dfa9c3b56acb15/lib/nettle/gost/kuztable.h#L73) as a constant 2-dimensional array (`static const uint8_t kuz_key_table[32][16]`) and the above access never involves out-of-bounds.  Perhaps #49948 might be related.

**Comment by t184256** [comment]
Extracted out and trimmed down:

```
int table[2][2] = { {1, 2}, {3, 4} };
int f(const int* c) { return c[0]; }
int main() { return f(table[1]); }
```

```
test.c:2:23: warning: Undefined or garbage value returned to caller [core.uninitialized.UndefReturn]
int f(const int* c) { return c[0]; }
                      ^~~~~~~~~~~
```

Curiously, passing `table[0]` or dropping the `const` modifier makes the warning go away.
